### PR TITLE
fix #5 - add formality/split_sentences

### LINKED ISSUE #5: add support for formality
The [API](https://www.deepl.com/docs-api/translating-text/) knows a lot more settings than your addon can configure. It would be helpful to be able to configure the formality of the translated text, e.g.`formality=less`

```
$ curl https://api-free.deepl.com/v2/translate -d auth_key=xxx -d "text=Hello, how are you?" -d "target_lang=DE" -d "formality=less"

{"translations":[{"detected_source_language":"EN","text":"Hallo, wie geht es dir?"}]}
```
VS
```
$ curl https://api-free.deepl.com/v2/translate -d auth_key=xxx -d "text=Hello, how are you?" -d "target_lang=DE" -d "formality=more"

{"translations":[{"detected_source_language":"EN","text":"Hallo, wie geht es Ihnen?"}]}
```

### PR DESCRIPTION
**Pull request by luto**
see #5 for details.

### CODE REVIEW

File: src/deepl.ts [hunk 1]
```diff
@@ -74,7 +76,7 @@ http.interceptors.response.use(
       throw e;
     }
 
-    const exception = DeepLException.createFromStatusCode(e.response.status);
+    const exception = DeepLException.createFromResponse(e.response.status, e.response.data.message);
```
[the remark this answers is not in the document]
Comment (luto) [reply src/deepl.ts:79]: I was trying to keep the parameters of the function simple by not passing the entire, complicated request object in there. How about `createFromStatusCodeAndMessage`?

**Comment by luto** [comment]
> Could you please use camelCase? Please run eslint to see possibles code style issues.

Of course, fixed! Sorry, I am not very familiar with typescript tooling.

File: src/deepl.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (soerenuhrbach) [reply src/deepl.ts:79]: Well, it's fine!

[the remark this answers is not in the document]
Comment (luto) [reply src/deepl.ts:79]: ok, I will get that changed then.

**Comment by luto** [comment]
squashed the fixup commits. From my point of view this is ready for merging. Let me know, if there is something else!

**Review by soerenuhrbach — APPROVED** [review]

**Comment by soerenuhrbach** [comment]
Thanks for contributing!

**Comment by luto** [comment]
thanks for the quick and constructive review! :tada:

**Pull request merged**